07-javascript.js

```diff
@@ -143,6 +163,25 @@ assert(p.hasOwnProperty("name"))
 // Penalty: -3 layout, -3 naming
 
 //START
+function bugs(callback) {
+
+  callback = function(){
+    String.prototype.sup = function(){
+      return `<sup>${this}</sup>`
+    }
+  }
+
+  try{
+    String.prototype.sup = function(){
+      return `What's up, ${this}?`
+    }
+  }
+  
+  finally{
+    callback()
```

Comment:
The idea of the try/catch is to handle the case where the callback raises an exception, so that the original value of `sup()` can be restored.

You never see an exception because you never all the actual callback..